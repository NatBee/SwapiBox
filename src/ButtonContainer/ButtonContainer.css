.btn-container {
  background-color: magenta;
  height: 75px; }
