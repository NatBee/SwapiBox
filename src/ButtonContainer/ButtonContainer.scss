.btn-container {
  background-color: #8E7842;;
  height: 35px;
  display: flex;
  justify-content: space-evenly;
  padding: 30px 0px;
}

button {
  font-family: 'Anton', sans-serif;
  background-color: #0C0A0A;
  color: #F7F4DB; 
  width: 100px;
  height: 35px;
  border-radius: 5%;
}

button:hover {
  background-color: #F7F4DB;
  color: #0C0A0A;
}

button:focus {
  background-color: #F7F4DB;
  color: #0C0A0A;
}