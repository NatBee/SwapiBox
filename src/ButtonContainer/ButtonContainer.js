import React from 'react';
import './ButtonContainer.css';

const ButtonContainer = ({handleClickPeople, handleClickPlanets, handleClickVehicles, handleClickFavorites, favorites}) => {

  const numberOfFavorites = favorites.length;

  return (
    <section className="btn-container">
      <button className="people" onClick={ () => handleClickPeople()}>
        People
      </button>
      <button className="planet" onClick={ () => handleClickPlanets()}>
        Planet
      </button>      
      <button className="vehicle" onClick={ () => handleClickVehicles()}>
        Vehicle
      </button>
       <button className="favorites" onClick={ () => handleClickFavorites()}>
        View Favorites 
        <div>
          {numberOfFavorites}
        </div>
      </button>
    </section>
  )
}

export default ButtonContainer