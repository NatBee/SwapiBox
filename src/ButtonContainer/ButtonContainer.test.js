/* eslint-disable */ 
import React from 'react';
import ReactDOM from 'react-dom';
import ButtonContainer from './ButtonContainer.js';
import { shallow } from 'enzyme';

it('should match the snapshot', () => {
  const renderedComponent = shallow(<ButtonContainer />);
  
  expect(renderedComponent).toMatchSnapshot();
})