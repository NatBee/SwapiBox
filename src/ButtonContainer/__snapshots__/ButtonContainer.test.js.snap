// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="people"
          onClick={[Function]}
>
          People
</button>,
        <button
          className="planet"
          onClick={[Function]}
>
          Planet
</button>,
        <button
          className="vehicle"
          onClick={[Function]}
>
          Vehicle
</button>,
      ],
      "className": "btn-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planet",
          "className": "planet",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planet",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicle",
          "className": "vehicle",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicle",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="people"
            onClick={[Function]}
>
            People
</button>,
          <button
            className="planet"
            onClick={[Function]}
>
            Planet
</button>,
          <button
            className="vehicle"
            onClick={[Function]}
>
            Vehicle
</button>,
        ],
        "className": "btn-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planet",
            "className": "planet",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planet",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicle",
            "className": "vehicle",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicle",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
